### PR TITLE
Alexa Port

### PR DESCRIPTION
**Pull request by tlfrd**
This ports the Google Home game to Alexa, adding extra conversation logic to comply with Alexa skill standards.

It uses the same game class as the Google Home game, so high scores will be consistent.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ app.use(userUUIDMiddleware);
 app.use('/', require('./routes/index'));
 app.use('/voice', basicAuth, require('./routes/voice'));
 app.use('/interface', require('./routes/interface'));
+app.use('/alexa', require('./routes/voice-alexa').router);
```
Comment (Lily2point0) [inline app.js]: find a way to put `basicAuth` in front of the Alexa route

File: routes/voice-alexa.js [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+const debug = require('debug')('correlations-game:routes:voice');
+const express = require('express');
+const router = express.Router();
+const games = require('../bin/lib/game');
+const responses = require('../responses/content');
+const Alexa = require('alexa-sdk');
+
+const APP_ID = process.env.APP_ID;
+
+const spoor = require('../bin/lib/log-to-spoor');
```
Comment (Lily2point0) [inline routes/voice-alexa.js]: Ensure spoor tracking matches the latest that's on `master` for the Google route

File: routes/voice-alexa.js [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+function checkGuess(sessionId, guessValue, currentQuestion, callback) {
+    let guess = guessValue;   
+
+    getExpectedAnswers(sessionId)
+    .then(data => {            
+        const answers = data.answersReturned;
+        const seed = data.seedPerson;
+        
+        let expectedAnswers;
+        if (typeof answers[0] === 'string' || answers[0] instanceof String) {
```
Comment (Lily2point0) [inline routes/voice-alexa.js]: What are those 2 conditions for?

File: routes/voice-alexa.js [hunk 4]
```diff
@@ -0,0 +1,498 @@
...
+        fail: () => {
+            // Fail with internal server error
+            response.sendStatus(500);
+        },
+        succeed: data => {
+            response.send(data);
+        }
+    };
+
+    const alexa = Alexa.handler(request.body, context);
```
Comment (Lily2point0) [inline routes/voice-alexa.js]: Does it need to be duplicated here as it's already in `handler`?

File: routes/voice-alexa.js [hunk 5]
```diff
@@ -0,0 +1,498 @@
...
+});
+
+const helpStateHandlers = Alexa.CreateStateHandler(GAME_STATES.HELP, {
+    'helpTheUser': function () {
+        const sessionId = this.event.session.sessionId;
+        
+        const helpBody = games.check(sessionId)
+        .then(gameIsInProgress => {
+            let helpResponse = responses.help(gameIsInProgress).ssml;
+            helpResponse = removeSpeakTags(helpResponse);
```
Comment (Lily2point0) [inline routes/voice-alexa.js]: use the `.speech` property from the the response, avoids having to remove `<speak>` tags. Or ar the `<break>` tags supported by Alexa?

Comment (tlfrd) [reply routes/voice-alexa.js]: Alexa automatically adds <speak> tags, so unfortunately this is necessary

File: routes/voice-alexa.js (hunk 4 above)
Comment (tlfrd) [reply routes/voice-alexa.js]: This has been resolved

**Pull request merged**